code/game/gamemodes/umbrage/umbrage_objects.dm

```diff
@@ -0,0 +1,156 @@
+//Dark bead: Formed by the Devour Will ability. See umbrage_abilities.dm for more details.
+/obj/item/weapon/dark_bead
+	name = "dark bead"
+	desc = "A glowing black orb. It's fading fast."
+	icon_state = "dark_bead"
+	item_state = "ratvars_flame"
+	flags = NODROP
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | UNACIDABLE | INDESTRUCTIBLE
+	w_class = 5
+	var/eating = 0 //If we're devouring someone's will
+	var/datum/action/innate/umbrage/devour_will/linked_ability //The ability that keeps data for us
+
+/obj/item/weapon/dark_bead/New()
+	..()
+	animate(src, alpha = 30, time = 30)
+	spawn(30)
+		if(!eating)
+			loc << "<span class='warning'>You were too slow! [src] faded away.</span>"
+			qdel(src)
+
+/obj/item/weapon/dark_bead/attack(mob/living/carbon/L, mob/living/user)
+	if(!is_umbrage(user.mind) || eating || L == user) //no eating urself ;)))))))
+		return
+	var/datum/umbrage/U = linked_ability.linked_umbrage
+	if(!L.health)
+		user << "<span class='warning'>[L] is too weak to drain.</span>"
+		return
+	for(var/V in linked_ability.victims)
```

Comment:
learn to love them.
they're your friend.
all they want to do is give you lovely O(1) constant time access, compared to ugly O(N) time access.